ChurchCRM/CRM
# Bugfix/4861 deposit strict mode

#### What's this PR do?
fixes the errors on the deposit slip screen when PHP strict mode is enabled.

#### Screenshots (if appropriate)
Before:
![image](https://user-images.githubusercontent.com/11679900/60773177-bc56f280-a0cf-11e9-8c2e-06972e1e0d1d.png)

After:
![image](https://user-images.githubusercontent.com/11679900/60773179-c842b480-a0cf-11e9-8727-72bf70eb9f6e.png)



#### What Issues does it Close?

Closes #4861


#### Any background context you want to provide?
Overriding the base ORM methods should be done with the same method parameters, otherwise PHP will throw a warning in strict mode.  This PR ensures that the method overrides have the same parameters, and that the overridden methods pass the parameters to the parent methods (if applicable)

#### How should this be manually tested?
Ensure that deposits can be created.
Ensure that pledges can be added to a deposit.
Ensure that a deposit can be delete if not yet closed.
Ensure that a deposit cannot be deleted if closed.
Ensure that the family name appears correctly in the "Payments on this Deposit Slip" data table.


#### How should the automated tests treat this?
Automated tests should exercise the deposit system.

Additionally, automated tests *should* run in strict mode, however I don't think that's feasible at this time.

#### Questions:
- Is there a related website / article to substantiate / explain this change?

- Does the development wiki need an update?

- Does the user documentation wiki need an update?
  -  [x ] No

- Does this add new dependencies?
  -  [ x] No

- Does this need to add new data to the demo database
  -  [ x] No
